packages/forms/src/UIForm/fields/NestedListView/NestedListView.utils.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Build the ListView component props to match current state's values
+ * @param {Array} items
+ * @param {Object} value
+ * @param {String} searchCriteria
+ * @param {Array} toggledChildren
+ * @returns {Object}
+ */
+export function getItemsProps(items, value, searchCriteria, toggledChildren) {
+	const displayedItems = items.reduce((finalItems, item) => {
```

Comment:
Reducing an array to an array seems the operations could have been done with some filters/maps chained.
It will isolate more clearly what's the filtering conditions and what's the calculated values returned and also remove all the manual array declaration and pushing call.

Example :
```javascript
return items
    .map() // Transform to an object with `item` and an other prop `children` based on the `searchCriteria`
    .filter() // Filter on children length
    .map() // Only the final item
```